### PR TITLE
Run regresssion tests on LGTM

### PR DESCRIPTION
**Pull request by pranav-vempati**
This PR modifies the CI to only run for PR commits, and runs the entire CI pipeline(unit and regression tests) only when the LGTM label is added to the PR.

### CODE REVIEW

**Comment by pranav-vempati** [comment]
LGTM

**Comment by pranav-vempati** [comment]
Test1

**Comment by pranav-vempati** [comment]
Test2

**Comment by pranav-vempati** [comment]
Test3

**Comment by pranav-vempati** [comment]
Test4

**Comment by pranav-vempati** [comment]
LGTM

**Comment by pranav-vempati** [comment]
#LGTM

**Comment by pranav-vempati** [comment]
Test19.

**Comment by pranav-vempati** [comment]
Comment.

**Comment by dylan-copeland** [comment]
@pranav-vempati will it be possible to trigger the pipeline multiple times in a PR, by removing and adding the LGTM label back? For example, if LGTM is added and the pipeline runs, then another commit is made, can we run the pipeline again by removing and adding LGTM?

**Review by chldkdtn — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
   pull_request:
     branches:
       - master
+  label:
+    types: [created]
 jobs:
   linux:
     runs-on: ubuntu-latest
     steps:
+
```
Comment (siuwuncheung) [inline .github/workflows/ci.yml:12]: Seems like a redundant blank line.

Comment (pranav-vempati) [reply .github/workflows/ci.yml]: Fixed.

**Review by siuwuncheung — APPROVED** [review]

**Comment by dylan-copeland** [comment]
@pranav-vempati does this PR ensure that (1) tests are run on every commit made after the LGTM label has been added, or (2) tests are run just once on the event that the LGTM label is added?

**Comment by pranav-vempati** [comment]
@dylan-copeland The answer is actually 1).  The regression tests run in two cases: when the LGTM label is added, and for every subsequent PR commit while the LGTM label is still present. If the regression tests should not run, the LGTM label can be removed.

**Pull request merged**